qui/updater/summary_page.py

```diff
@@ -0,0 +1,370 @@
+
+        self.restart_list.connect("row-activated",
+                                  self.on_checkbox_toggled)
+        self.app_vm_list: Gtk.ListStore = self.builder.get_object(
+            "restart_list_store")
+        restart_checkbox_column: Gtk.TreeViewColumn = self.builder.get_object(
+            "restart_checkbox_column")
+        restart_checkbox_column.connect("clicked",
+                                        self.on_header_toggled)
+        restart_header_button = restart_checkbox_column.get_button()
+        restart_header_button.connect('realize', pass_through_event_window)
+
+        self.head_checkbox_button: Gtk.CheckButton = self.builder.get_object(
+            "restart_checkbox_header")
+        self.head_checkbox_button.set_inconsistent(True)
+        self.head_checkbox_button.connect(
+            "toggled", self.on_header_toggled)
+        self.head_checkbox = RestartHeaderCheckbox(
+            self.head_checkbox_button, self.next_button)
+
+        self.summary_list: Gtk.TreeView = self.builder.get_object(
+            "summary_list")
+        self.summary_list.connect("row-activated", back_by_row_selection)
+
+    @disable_checkboxes
+    def on_checkbox_toggled(self, _emitter, path, *_args):
+        """Handles (un)selection of single row."""
+        if path is None:
+            return
+
+        self.list_store.invert_selection(path)
+        self.refresh_buttons()
+
+    def refresh_buttons(self):
+        """Refresh additional info column and finish button info."""
+        for row in self.list_store:
+            row.refresh_additional_info()
+        selected_num = sum(
+            row.selected for row in self.list_store)
+        if selected_num == 0:
+            self.head_checkbox.state = HeaderCheckbox.NONE
+        elif selected_num == len(self.list_store):
+            self.head_checkbox.state = HeaderCheckbox.ALL
+        else:
+            self.head_checkbox.state = HeaderCheckbox.SELECTED
+        self.head_checkbox.set_buttons(selected_num)
+
+    @disable_checkboxes
+    def on_header_toggled(self, _emitter):
+        """Handles clicking on header checkbox.
+
+        Cycle between selection from appvms which templates was updated :
+         <1> only sys-vms
+         <2> sys-vms + other appvms but without excluded in settings
+         <3> all appvms
+         <4> no vm. (nothing)
+
+        If the user has selected any vms that do not match the defined states,
+        the cycle will start from (1).
+        """
+        on_head_checkbox_toggled(
+            self.list_store, self.head_checkbox, self.select_rows)
+
+    @property
+    def is_populated(self) -> bool:
+        """Returns True if restart list is populated."""
+        return self.list_store is not None
+
+    @property
+    def is_visible(self):
+        """Returns True if page is shown by stack."""
+        return self.stack.get_visible_child() == self.page
+
+    def show(
```

Comment:
I've got the following:
![oopsie](https://user-images.githubusercontent.com/7242298/221614252-0f587c32-b307-4f08-a110-5236651a38fd.png)
I'm 99% sure this blank qube is a running `personal` qube.